packages/kit/src/packaging/index.js

```diff
@@ -86,19 +82,20 @@ export async function make_package(config, cwd = process.cwd()) {
 
 		write(path.join(cwd, config.kit.package.dir, out_file), out_contents);
 
-		if (!user_defined_exports && exports_filter(file)) {
+		if (exports_filter(file)) {
 			const entry = `./${out_file.replace(/\\/g, '/')}`;
 			const key = entry.endsWith('/index.js') ? entry.slice(0, -'/index.js'.length) : entry;
-			pkg.exports[key] = entry;
+			generated[key] = entry;
 		}
 	}
 
-	const main = pkg.exports['./index.js'] || pkg.exports['./index.svelte'];
-
-	if (!user_defined_exports && main) {
-		pkg.exports['.'] = main;
+	if (!pkg.exports) {
+		pkg.exports = generated;
+	} else if (config.kit.package.exports.behavior === 'replace') {
+		pkg.exports['./package.json'] = './package.json';
+	} else if (config.kit.package.exports.behavior === 'merge') {
+		pkg.exports = { ...generated, ...pkg.exports };
```

Comment:
This _might_ not be enough for some, they could expect a deep merge. Not sure what the best default here would be.